diff --git a/Simbody/Visualizer/simbody-visualizer/CMakeLists.txt b/Simbody/Visualizer/simbody-visualizer/CMakeLists.txt
index be78c913..80f2c7fa 100644
--- a/Simbody/Visualizer/simbody-visualizer/CMakeLists.txt
+++ b/Simbody/Visualizer/simbody-visualizer/CMakeLists.txt
@@ -38,14 +38,14 @@ endif()
 
 if(OPENGL_FOUND AND SIMBODY_HAS_GLUT)
 
+if(UNIX AND NOT APPLE)
+    link_directories(/usr/lib64)
+endif()
+
 add_executable(${GUI_NAME} 
     simbody-visualizer.cpp lodepng.cpp lodepng.h
     ${GLUT32_HEADERS}) # only on Windows
 
-if(NOT WIN32)
-    target_include_directories(${GUI_NAME} PRIVATE ${GLUT_INCLUDE_DIR})
-endif()
-
 # If building as debug, append the debug postfix to the name of the executable.
 # CMAKE_DEBUG_POSTFIX only affects non-executable targets, but we use its value
 # to set the postfix for this executable.
@@ -76,7 +76,22 @@ else()
     target_link_libraries(${GUI_NAME} ${TEST_STATIC_TARGET})
 endif()
     
-target_link_libraries(${GUI_NAME} ${GLUT_LIBRARIES} ${OPENGL_LIBRARIES})
+message("DEBUG GLUT_LIBRARIES ${GLUT_LIBRARIES}")
+message("DEBUG OPENGL_LIBRARIES ${OPENGL_LIBRARIES}")
+message("DEBUG ${CMAKE_CXX_IMPLICIT_LINK_DIRECTORIES}")
+if(CMAKE_VERSION VERSION_LESS 3.8)
+    if(NOT WIN32)
+        target_include_directories(${GUI_NAME} PRIVATE ${GLUT_INCLUDE_DIR})
+    endif()
+    target_link_libraries(${GUI_NAME} ${GLUT_LIBRARIES} ${OPENGL_LIBRARIES})
+else()
+    if(WIN32)
+        target_link_libraries(${GUI_NAME} ${GLUT_LIBRARIES} ${OPENGL_LIBRARIES})
+    else()
+        target_link_libraries(${GUI_NAME} PRIVATE OpenGL::GL OpenGL::GLU GLUT::GLUT)
+    endif()
+endif()
+
 
 # SIMBODY_VISUALIZER_INSTALL_DIR is set in the root CMakeLists.txt
 install(TARGETS ${GUI_NAME}
